src/utils/CancelableRequest.js

```diff
@@ -21,36 +21,30 @@
  *
  */
 
-import axios from '@nextcloud/axios'
-
 /**
  * Creates a cancelable axios 'request object'.
  *
  * @param {Function} request the axios promise request
  * @return {object}
  */
-const CancelableRequest = function(request) {
-	/**
-	 * Generate an axios cancel token
-	 */
-	const CancelToken = axios.CancelToken
-	const source = CancelToken.source()
+const CancelableRequest = function (request) {
+	const controller = new AbortController()
 
 	/**
 	 * Execute the request
 	 *
 	 * @param {string} url the url to send the request to
 	 * @param {object} [options] optional config for the request
 	 */
-	const fetch = async function(url, options) {
+	const fetch = async function (url, options) {
 		return request(
 			url,
-			Object.assign({ cancelToken: source.token }, { options })
+			{ ...options, signal: controller.signal }
```

Comment:
Ok to use the newer object spreading syntax?

Reply:
Yes please :D